### ISSUE TITLE
Can't find `TextEncoder` in tests after 2.2.2 release

### ISSUE THREAD

**Issue by karol-f** [Maggi64/moderndash#60]
Can't find `TextEncoder` in Jest tests after 2.2.2 release:

```
Test suite failed to run

    ReferenceError: TextEncoder is not defined

    > 1 | import { merge } from 'moderndash';
```

What is strange it want to get it from `src` folder:
```
at Object.<anonymous> (../../node_modules/moderndash/src/crypto/hash.ts:5:21)
```

I can see only `src` comment in the dist source code:
```
// src/crypto/hash.ts
var textEncoder = new TextEncoder();
```

**Comment by karol-f** [comment]
Using it in `setupFilesAfterEnv: ['<rootDir>/jest-setup.ts']` file worked for me. I just wonder if we really need `TextEncoder` in Node environment (e.g. in tests or Next.js on server-side). 

This did a trick for me, works in both environments:

```
var textEncoder = typeof TextEncoder === 'function' ? new TextEncoder() : undefined;
```

**Comment by Maggi64** [comment]
Sounds good 👍  I might change the function to only create the TextEncoder object when actually needed.
I need to check benchmarks what impact this would have though.
Without tree-shaking (i.e. in tests) TextEncoder gets called without ever getting used.

**Comment by karol-f** [comment]
> Without tree-shaking (i.e. in tests) TextEncoder gets called without ever getting used.

I will just add that `tree-shaking` is working only in production builds (as it's computation heavy), so I assume that in dev mode it will break in e.g. Next.js.

**Comment by Maggi64** [comment]
@karol-f Fixed in #61 2.2.3

TextEncoder is only created when its needed in `hash`

You can close this issue if it works for you 👍

**Comment by karol-f** [comment]
Thank you!
